packages/bvaughn-architecture-demo/src/hooks/useDismissNag.ts

```diff
@@ -0,0 +1,25 @@
+import { useContext } from "react";
+
+import { GraphQLClientContext } from "../contexts/GraphQLClientContext";
+import { SessionContext } from "../contexts/SessionContext";
+import { Nag } from "../graphql/types";
+import { dismissNag } from "../graphql/User";
+
+export function useDismissNag() {
+  const { accessToken, currentUserInfo } = useContext(SessionContext);
+  const graphQLClient = useContext(GraphQLClientContext);
+
+  return (nag: Nag) => {
+    if (!accessToken || !currentUserInfo) {
+      return;
+    }
+
+    const { id, nags } = currentUserInfo;
+
+    if (!nags || nags.includes(nag) || !id) {
+      return;
+    }
+
+    dismissNag(graphQLClient, accessToken, nag);
```

Comment:
I think _maybe_ the reason this works differently for the old console is that the mutation was configured to also refetch user info:
```js
const [dismissNag, { error }] = useMutation<DismissNag, DismissNagVariables>(DISMISS_NAG, {
  refetchQueries: ["GetUser"],
});
```

The new architecture stuff doesn't use Apollo, just plain GraphQL, so there's nothing that tells the legacy app (with Apollo) that it needs to refresh user info.

We can address this as part of the follow up issue though! 😄

Reply:
Oh gosh, that makes sense why it's not updating 🤦 sounds good!